src/ape_ethereum/transactions.py

```diff
@@ -61,9 +60,6 @@ def serialize_transaction(self) -> bytes:
         signature = (self.signature.v, to_int(self.signature.r), to_int(self.signature.s))
         signed_txn = encode_transaction(unsigned_txn, signature)
 
-        if self.sender and EthAccount.recover_transaction(signed_txn) != self.sender:
```

Comment:
I am not opposed to deleting this, after some careful thought.
I am wondering if there is another way, in case something is actually relying on the signature error (which I kind of doubt, but just being cautious).

**Are we able to detect if the sender is an impersonated account here?** If so, maybe we can skip this check.

Reply:
maybe `if isinstance(self.sender, (ContractInstance, ImpersonatedAccount, AddressType))`